schemas/iso19139/src/main/plugin/iso19139/layout/tpl-rdf.xsl

```diff
@@ -427,26 +427,45 @@
     <!-- xpath: gmd:identificationInfo/*/gmd:language/gmd:LanguageCode/@codeListValue -->
 
 
+    <!-- dct:license content -->
     <!-- "The license under which the dataset is published and can be reused." -->
-    <xsl:for-each select="gmd:resourceConstraints/gmd:MD_LegalConstraints/*/gmd:MD_RestrictionCode[@codeListValue!='otherRestrictions']">
-      <dct:license>
-        <xsl:value-of select="@codeListValue"/>
-      </dct:license>
-    </xsl:for-each>
-    <xsl:for-each
-      select="gmd:resourceConstraints/gmd:MD_LegalConstraints/gmd:otherConstraints/gco:CharacterString">
-      <dct:license>
-        <xsl:value-of select="."/>
-      </dct:license>
+    <xsl:for-each select="gmd:resourceConstraints/gmd:MD_LegalConstraints/gmd:useConstraints">
+      <xsl:choose>
+        <xsl:when test="gmd:MD_RestrictionCode[@codeListValue!='otherRestrictions']">
+          <dct:license>
+            <xsl:value-of select="@codeListValue"/>
+          </dct:license>
+        </xsl:when>
+        <xsl:otherwise>
+            <xsl:call-template name="legalOtherConstraints">
+              <xsl:with-param name="ocnode"><xsl:copy-of select="./following-sibling::gmd:otherConstraints[1]"/></xsl:with-param>
```

Comment:
Thanks @fxprunayre for ths analysis. This can obviously be improved. I'm really not an expert on this topic, so my next question may be trivial. But in the second MD_LegalConstraints block, both useLimitation and useConstraints are defined. Is it expected pattern ? How do you then know which otherConstraints apply to which tag ?

As an alternative, we can say that every otherConstraints tag that appears in the same MD_LegalConstraints block as an useConstraints tag are defined as licence, it seems fair to me. That would put the responsibility on the metadata editor to format it properly if he wants only 1 dct:license tag (to comply with geodcat-ap) and should not break anything already existing

Reply:
But on the example metadata you give, at least the No limitations to public access mention should not be listed as dct:license but rather dct:accessRights, right ?

Reply:
@fxprunayre @josegar74 what do you think ? Should I bring some modifications to this ? I there a chance I can get this PR to be accepted ?

Reply:
I think we should avoid to lose information - this change only take into account the first otherConstraints.

Maybe we should rely more on https://github.com/SEMICeu/iso-19139-to-dcat-ap/blob/master/iso-19139-to-dcat-ap.xsl ? 

Testing quickly, the SEMICeu template for constraints looks to preserve all elements and dispatch information properly between license and accessRights 
![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/30a96759-de14-468a-bcf8-d289bab2ba24)
Not sure it will be fine for the data.gouv.fr harvester - can you give a try?

In Belgium, there are also some analysis ongoing to improve the ISO to DCAT mapping.

Reply:
Yes, sure, I'll try this afternoon and keep you posted. Looks like a fine idea